### DISCUSSION TITLE
How to get rid of "This branch has not been deployed" banner

### DISCUSSION THREAD

**Discussion by ItsDrike** [community/community#46375] [Pull Requests] [answered]
### Select Topic Area

Question

### Body

I'm seeing this This branch has not been deployed banner with "No deployments" description in all of my PRs. What's causing this? How can I get rid of it?

![image](https://user-images.githubusercontent.com/20902250/216789753-38b75967-61a9-4986-8b4b-09112585537b.png)

See: https://github.com/py-mine/mcproto/pull/26

I'm pretty sure that this was cause by me messing with the project settings and creating a new environment:
![image](https://user-images.githubusercontent.com/20902250/216790934-7ef0dcf4-2581-49c1-bc6d-b5c45eac92d0.png)

As I was trying to separate the secrets for different environments for workflows, but I've since removed this environment (the screenshot is from a mock one that I've just created with same settings (iirc) as the original one).

What's interesting is that this environment that I had was named "external", and when I go to protected branches, and check "Require deployments to succeed before merging" (I don't have this on currently, I'm just showing what appears when I select it), in the deployment environments, I can see the "external" name.
![image](https://user-images.githubusercontent.com/20902250/216791133-5c04abd6-f3eb-4786-bdb6-222e21c51c1f.png)

So I probably just need to delete this deployment environment, but how? I've already removed the environment in settings a long time ago. Where is this comming from?
![image](https://user-images.githubusercontent.com/20902250/216791203-ebe43ee3-cc91-4e88-af44-7f43e96b54f0.png)

This is what I can see in https://github.com/py-mine/mcproto/deployments
![image](https://user-images.githubusercontent.com/20902250/216791274-dfd01a1b-587a-4cea-acbc-a423f7f82ae5.png)

**Comment by ItsDrike** [comment, marked as answer]
Ok, so I managed to solve this, however it was not intuitive at all!

Following the GitHub API docs: https://docs.github.com/en/rest/deployments/deployments?apiVersion=2022-11-28#delete-a-deployment

There is a way to send an API request to delete a specific deployment, I fist sent this request:
```bash
curl \
  -H "Accept: application/vnd.github+json" \
  -H "Authorization: Bearer <MY_GITHUB_TOKEN>"\
  -H "X-GitHub-Api-Version: 2022-11-28" \
  https://api.github.com/repos/py-mine/mcproto/deployments
```

Which returned a JSON, containing the list of all deployments for the repo (there were 2). From that, I sent a delete request to both of these deployments:
```bash
curl \
  -X DELETE \
  -H "Accept: application/vnd.github+json" \
  -H "Authorization: Bearer <MY_GITHUB_TOKEN>"\
  -H "X-GitHub-Api-Version: 2022-11-28" \
  https://api.github.com/repos/py-mine/mcproto/deployments/<DEPLOYMENT_ID>
```

After that, the banners disappeared. Though this should really have been much more intuitive. Any place to report this as a bug/request for improvement?

**Reply by queenofcorgis** [reply]
Hi there - feel free to leave some product feedback in our [API and Webhooks category](https://github.com/orgs/community/discussions/categories/api-and-webhooks?discussions_q=sort%3Atop+created%3A%3E%3D2023-01-22+is%3Aanswered+category%3A%22API+and+Webhooks%22). 

cc: @ghostinhershell as she oversees the category 😄

**Reply by rwojsznis** [reply]
Pfew, thanks for this! For folks coming from search; to wipe all deployments:

```bash
gh api /repos/<ORG>/<REPO>/deployments --jq=".[].id" | \
xargs -I {} gh api --method DELETE -H "Accept: application/vnd.github+json" -H "X-GitHub-Api-Version: 2022-11-28" "/repos/<ORG>/<REPO>/deployments/{}"
```

(not sure if the accept/headers are needed tbh, seems like now it's explicitly listed in the docs at time of writing this)

**Comment by soda92** [comment]
Well I migrated from Vercel to Firebase hosting and this message troubles me.

Here is a Windows version of the script:
```pwsh
$User = "USER"
$Repo = "REPO"
gh api /repos/$User/$Repo/deployments --jq=".[].id" `
| ForEach-Object {
  gh api --method DELETE -H "Accept: application/vnd.github+json" `
    -H "X-GitHub-Api-Version: 2022-11-28" `
    "/repos/$User/$Repo/deployments/$_"  
}
```
